server/workflowtemplate/workflow_template_server.go

```diff
@@ -69,15 +71,82 @@ func (wts *WorkflowTemplateServer) getTemplateAndValidate(ctx context.Context, n
 
 func (wts *WorkflowTemplateServer) ListWorkflowTemplates(ctx context.Context, req *workflowtemplatepkg.WorkflowTemplateListRequest) (*v1alpha1.WorkflowTemplateList, error) {
 	wfClient := auth.GetWfClient(ctx)
-	options := &v1.ListOptions{}
+	k8sOptions := &v1.ListOptions{}
+
 	if req.ListOptions != nil {
-		options = req.ListOptions
+		k8sOptions = req.ListOptions
 	}
-	wts.instanceIDService.With(options)
-	wfList, err := wfClient.ArgoprojV1alpha1().WorkflowTemplates(req.Namespace).List(ctx, *options)
+
+	// Save the original Continue and Limit for custom filtering.
+	resourceVersion := k8sOptions.Continue
+	limit := k8sOptions.Limit
+
+	// Search whole with Limit 0.
+	// Reset the Continue "" to prevent Kubernetes native pagination.
+	// Kubernetes api will search for all results without limit and pagination.
+	k8sOptions.Continue = ""
+	k8sOptions.Limit = 0
```

Comment:
Would it be possible to stick with the native pagination when search functionality is not used?

Reply:
I think maybe it is possible. I will try to implement it like that.

Reply:
I uploaded commit about sticking with native pagination when search functionality is not used. 392222bd097eaa2374de08eb59f9e5feb78c2c61

If this PR is reflected, we plan to implement the same namePattern function on other pages as well. I will consider having `cursorPaginatonByResourceVersion` common in one place and one function.